cmd/misc/main.go

```diff
@@ -1536,3 +1549,354 @@ func reExportSyncCommittee(rpcClient rpc.Client, p uint64, dryRun bool) error {
+		}
+		logrus.Infof("exported slot %v", slot)
+	}
+
+	for epoch := range affectedEpochs {
+		// update epoch status
+		epochParticipationStats, err := clClient.GetValidatorParticipation(epoch)
+		if err != nil {
+			return fmt.Errorf("error getting validator participation for epoch %v: %w", epoch, err)
+		}
+		err = db.UpdateEpochStatus(epochParticipationStats, tx)
+		if err != nil {
+			return fmt.Errorf("error updating epoch status for epoch %v: %w", epoch, err)
+		}
+		logrus.Infof("updated epoch status for epoch %v", epoch)
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return fmt.Errorf("error committing db tx: %w", err)
+	}
+
+	affectedDaysArr := []uint64{}
+	for day := range affectedDays {
+		affectedDaysArr = append(affectedDaysArr, day)
+	}
+	fmt.Printf("stats for these days must be recalculated: %v\n", affectedDaysArr)
+
+	return nil
+}
+
+// bids2330FixBlocks has been implemented for bids-2330
+func bids2330FixBlocks() error {
+	logrus.WithFields(logrus.Fields{"startBlock": opts.StartBlock, "endBlock": opts.EndBlock, "dry": opts.DryRun}).Infof("running command fixBlocks")
+
+	type ClEthV1ConfigSpecResponse struct {
+		Data map[string]string `json:"data"`
+	}
+
+	type ClEthV2BeaconBlocksBlockResponse struct {
+		Version             string `json:"version"`
+		ExecutionOptimistic bool   `json:"execution_optimistic"`
+		Finalized           bool   `json:"finalized"`
+		Data                struct {
+			Message struct {
+				Slot string `json:"slot"`
+			}
+		}
+	}
+
+	spec := &ClEthV1ConfigSpecResponse{}
+	err := utils.HttpReq(context.Background(), http.MethodGet, fmt.Sprintf("http://%s:%s/eth/v1/config/spec", utils.Config.Indexer.Node.Host, utils.Config.Indexer.Node.Port), nil, spec)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+	nodeDepositNetworkId, ok := spec.Data["DEPOSIT_NETWORK_ID"]
+	if !ok {
+		logrus.Fatal(fmt.Errorf("missing DEPOSIT_NETWORK_ID in spec from node"))
+	}
+	if fmt.Sprintf("%d", utils.Config.Chain.ClConfig.DepositNetworkID) != nodeDepositNetworkId {
+		logrus.Fatal(fmt.Errorf("config.DepositNetworkId != node.DepositNetworkId: %v != %v", utils.Config.Chain.ClConfig.DepositNetworkID, nodeDepositNetworkId))
+	}
+
+	start := opts.StartBlock
+	end := opts.EndBlock
+	if end == 0 {
+		end = math.MaxUint64
+	}
+
+	for i := start; i < end; i += 1000 {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+		defer cancel()
+		res := &ClEthV2BeaconBlocksBlockResponse{}
+		err := utils.HttpReq(ctx, http.MethodGet, fmt.Sprintf("http://%s:%s/eth/v1/beacon/blocks/finalized", utils.Config.Indexer.Node.Host, utils.Config.Indexer.Node.Port), nil, res)
+		if err != nil {
+			return fmt.Errorf("error getting finalized block: %w", err)
+		}
+
+		finalizedSlot, err := strconv.ParseUint(res.Data.Message.Slot, 10, 64)
+		if err != nil {
+			return fmt.Errorf("error parsing finalized slot: %w", err)
+		}
+		iStart := i
+		iEnd := i + 1000
+		if iEnd > finalizedSlot+1 {
+			iEnd = finalizedSlot + 1
+		}
```

Comment:
**Minor:** This should also include something like
```Go
if iEnd > end {
	iEnd = end
}
```

Because right now it could happen that `start` is something like `end - 3` and in that case end would still be `iEnd` would still result in `start + 1000`.